smart-contract-verifier/smart-contract-verifier/src/verify/solc_compiler_cli.rs

```diff
@@ -231,31 +233,79 @@ mod types {
         pub contracts: HashMap<String, OutputContract>,
     }
 
-    fn remove_path_from_contract_name(name: String) -> String {
+    #[derive(Debug, PartialEq, Eq, Clone)]
+    pub struct ConvertibleOutput<'a> {
+        pub output_json: OutputJson,
+        pub parent_dir: &'a Path,
+        pub file_names: Vec<PathBuf>,
+    }
+
+    fn split_file_and_contract_names(name: String) -> (Option<String>, String) {
         name.rsplit_once(':')
-            .map(|(_, name_cleared)| name_cleared.to_string())
-            .unwrap_or(name)
+            .map(|(file_name, contract_name)| {
+                (Some(file_name.to_string()), contract_name.to_string())
+            })
+            .unwrap_or((None, name))
     }
 
-    impl TryFrom<OutputJson> for solc::CompilerOutput {
+    impl TryFrom<ConvertibleOutput<'_>> for solc::CompilerOutput {
         type Error = SolcError;
 
-        fn try_from(output_json: OutputJson) -> Result<Self, Self::Error> {
-            let contracts = output_json
-                .contracts
-                .into_iter()
-                .map(|(name, output)| {
-                    let name = remove_path_from_contract_name(name);
-                    output.try_into().map(|contract| (name, contract))
-                })
-                .collect::<Result<HashMap<String, solc::Contract>, _>>()?;
+        fn try_from(convertible_output: ConvertibleOutput) -> Result<Self, Self::Error> {
+            // The easiest option - we always know the compiled contract name
+            let contracts = if convertible_output.file_names.len() == 1 {
+                debug_assert!(convertible_output.output_json.contracts.len() == 1);
+
+                let full_file_path = convertible_output.file_names[0].clone();
+                let file_name = full_file_path
+                    .strip_prefix(convertible_output.parent_dir)
+                    // must never be the case, as we derive that full_file_path during InputFiles derivation
+                    .map_err(|_| SolcError::Message("compiled file has unexpected prefix".into()))?
+                    .to_string_lossy()
+                    .to_string();
+
+                let (name, output) = convertible_output
+                    .output_json
+                    .contracts
+                    .into_iter()
+                    .next()
+                    .unwrap();
+                let (_, contract_name) = split_file_and_contract_names(name);
```

Comment:
why do u need file name if you dont use it here?

function `fn remove_path_from_contract_name(name: String) -> String` become `fn split_file_and_contract_names(name: String) -> (Option<String>, String)` for no reason 👀

Reply:
More or less that is just a subproduct of my attempt to support multiple files in cases when the compiler specifies those file names in the output. Though eventually we decided to give up multi-file support, I believe returning both file and contract names does no harm here and allows to use this method in the multi-file `else` branch if needed